.github/workflows/check.yml

```diff
@@ -0,0 +1,46 @@
+name: ✅ Tamagokill check
+on:
+  pull_request:
+    types: [opened, synchronize]
+
+jobs:
+  check-web:
+    runs-on: ubuntu-latest
+    steps:
+      - name: 🛎️ Checkout
+        uses: actions/checkout@v3
+
+      - name: 🏭 Use Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: current
```

Comment:
Should be a fixed version or a matrix so we do not have a breaking CI when the version change. Maybe have a matrix with current that throw a warning (but don't stop ci) so we can be ready for next version

Reply:
Is it not possible to use a global env var ? I don't know if it is possible to propegate them in github hooks. But like that we only have to change the version of a env variable ?

(Sorry if it makes no sense lol)

Reply:
Here's an example of a matrix

jobs:
  example_matrix:
    strategy:
      matrix:
        version: [10, 12, 14]
    steps:
      - uses: actions/setup-node@v4
        with:
          node-version: ${{ matrix.version }}
          
With that, you can try multiple node version at once on multiple jobs :)

Reply:
Go for your PR, I will open another one to demonstrate :)

Reply:
Using a matrix will use a lot of ressources, and we only need to maintain a single version of node at once.

Using a matrix would be useful if we were deploying a library for example, as it would be executed by multiple versions of node.

With this approach, to test the next version of node.js, you would need to create a new branch and change the NODE_VERSION default value.

Reply:
@Lauwed I'll set it up soon :)

Reply:
> Using a matrix will use a lot of ressources, and we only need to maintain a single version of node at once.

Agreed, we only need one version of node running

Reply:
Having a future version running would help me to decrease breaking changes when we will need to upgrade.

I'm fine with one version, of course, I have the future PHP version running for professional projects.

Open source projects have like 2000 hours of free CI, the resource used isn't above for a simple project on two versions.

Reply:
I just updated the PR, I was searching for a clean way to use env variables in github actions, a seamless way was to put the variables in a .env, which is a zero-config way to seed variables for both docker compose and github actions, but as it turns out it's a bad approach to commit this file in a public project https://dev.to/somedood/please-dont-commit-env-3o9h.

There is another way to define which version of node shall be used in a project by using engines.node in the package.json. It is used by default by actions/setup-node if defined.